src/tests/sys_tests.rs

```diff
@@ -68,5 +73,22 @@ pub mod test_sys {
         lindrustfinalize()
     } 
 
+    pub fn ut_lind_fork() {
+        // Since the fork syscall is heavily tested in relation to other syscalls
+        // we only perform simple checks for testing the sanity of the fork syscall
+        lindrustinit(0); 
+        let cage = interface::cagetable_getref(1); 
+        // Spawn a new child object using the fork syscall 
+        cage.fork_syscall(2); 
+        // Search for the new cage object with cage_id = 2
+        let child_cage = interface::cagetable_getref(2); 
+        // Assert the parent value is the the id of the first cage object
+        assert_eq!(child_cage.getpid_syscall(),1);
+        // Assert that the cage id of the child is the value passed in the original fork syscall
+        assert_eq!(child_cage.getuid(),2);
+        // Assert that the cwd is the same as the parent cage object
+        assert_eq!(child_cage.cwd.read(),cage.cwd.read())
+    }
+
```

Comment:
I suggest to add a test to test for error cases. e.g. what would happen if passing a pid that already exists?

Reply:
We don't handle the case of non-unique pids because forking is handled by the NaCl runtime - and it ensures that the parameter to fork is always unique.

Reply:
Thats something that definitely should be at a top level comment though.

Reply:
Added!